contracts/ExampleNFT.cdc

```diff
@@ -60,6 +61,16 @@ pub contract ExampleNFT: NonFungibleToken {
                     return MetadataViews.Royalties(
                         self.royalties
                     )
+                case Type<MetadataViews.NFTCollectionView>():
+                    return MetadataViews.NFTCollectionView(
+                        storagePath: ExampleNFT.CollectionStoragePath,
+                        publicPath: ExampleNFT.CollectionPublicPath,
+                        publicCollection: Type<&ExampleNFT.Collection{ExampleNFT.ExampleNFTCollectionPublic}>(),
+                        publicCollectionLinkedType: Type<&ExampleNFT.Collection{ExampleNFT.ExampleNFTCollectionPublic,NonFungibleToken.CollectionPublic}>(),
+                        createEmptyCollectionFunction: (fun (): @NonFungibleToken.Collection {
```

Comment:
I think can assign ‘ ExampleNFT.createEmptyCollection’ here

Reply:
That actually doesn't work, I don't know why though - this is what I get back with `ExampleNFT.createEmptyCollection`:
```
error: runtime error: invalid memory address or nil pointer dereference
```

Reply:
Would it be possible to have a path that is private and links to something that is a provider in adition to the standard restrictions? 

This would enable me to create a capability for say storefront or similar services from this information.

Reply:
Why are the paths in this example not including MetadataViews.ResolverCollection?

Reply:
@aishairzay this needs a bug report on cadence I guess, normally should work

Reply:
I will add metadataviews.resolver and nft.receiver @bjartek

I think I left them out while testing to shorten the strings in the tests 😛